client/landing/gutenboarding/stores/user/index.ts

```diff
@@ -0,0 +1,14 @@
+/**
+ * External dependencies
+ */
+import { User } from '@automattic/data-stores';
+
+/**
+ * Internal dependencies
+ */
+import config from '../../../../config';
+
+export const USER_STORE = User.register( {
```

Comment:
We'll need this for future requests. Is it something we could add to `client/landing/gutenboarding/constants.ts` or somewhere more appropriate?

Reply:
Don't like the idea of initialising runtime guff (like registering stores) in `constants.ts`. Maybe we should just change the data-store so it exports the key? What do you think about that @andrewserong? Was there a reason you wanted `.register` to return the key? It would mean we could do things like assign a random store key to avoid collisions?

Reply:
To be clear, if we exported the key from the data-store then the consuming code in a component might look like this:

```typescript
import { USER_STORE_KEY } from '@automattic/data-stores';

const { createAccount } = useDispatch( USER_STORE_KEY );
```

Reply:
Good question! I was mostly following the pattern used by the other stores. What I like about the pattern is that the consuming code is responsible for registering the store, instead of the data store package. So you can import anything from `packages/data-stores` but nothing gets registered until you specifically request it. The other thing we discussed was keeping the client id and secret out of `packages/data-stores` and `User.register( /* keys */ );` seemed like a natural place to do it. I'm open to other ideas, though!

Since the user store is going to be used potentially in a bunch of places, would it make sense to think of it as a part of the init process of loading Gutenboarding? So it could happen somewhere around `Window.AppBoot` in `client/landing/gutenboarding/index.tsx`? There's possibly a nice parallel there with `client/landing/login/index.js` which runs `const user = userFactory();` in `window.AppBoot`.

Reply:
Oh, wait, I might have misunderstood, we're just talking about the constant for the store key, aren't we? In that case, I think it makes sense to import it straight from `@automattic/data-stores`, but I'm wondering if the pattern used in the other packages was to ensure that you can only get the constant _after_ the store has been registered?

Reply:
Yeah I definitely like that the package doesn't have the side-effect of registering a store, just because it gets imported. I was thinking about make an `initStores()` function that gets called in `AppBoot`. Might be nicer that way because then the way and order that everything gets init'd will be deterministic.
It'll also give us a way to freshly init the stores in unit tests. So maybe I'll do that.

It's a toss up whether to export the store key from `@automattic/data-stores`. On the one hand I like the data hiding of having `register` return the key; other code isn't supposed to just know what that constant is. On the other hand the looser coupling of looking up a store at runtime using its key seems to be a feature of `@wordpress/data`, like it was designed that way i.e. you're supposed to be able to look up a store by its constant and fail gracefully if no one has registered it yet.

I'm inclined to just leave `@automattic/data-stores` the way it is since I don't know which is the right trade off for us.

Reply:
> Yeah I definitely like that the package doesn't have the side-effect of registering a store, just because it gets imported. I was thinking about make an initStores() function that gets called in AppBoot.

I like this idea, sounds neat and could be a helpful way of tracking the number of stores we're using in one place, too.

> I'm inclined to just leave @automattic/data-stores the way it is since I don't know which is the right trade off for us.

I'm not sure of the right trade off here either... I think the answer might become clearer to us once we have more use cases in Gutenboarding to guide us :)